syz-fuzzer/fuzzer.go

```diff
@@ -602,8 +661,41 @@ func triageInput(pid int, env *ipc.Env, inp Input) {
 	}
 }
 
-func execute(pid int, env *ipc.Env, p *prog.Prog, needCover, minimized, candidate bool, stat *uint64) []ipc.CallInfo {
+func executeHintSeed(pid int, env *ipc.Env, p *prog.Prog) {
+	if !kcovCompsEnabled {
+		return
+	}
+	// first execute the original program to dump comparisons from KCOV
+	// and to measure initial signals
+	info := execute(pid, env, p, false, true, false, false, false, &statExecHintSeeds)
+	// parse the comparisons and generate the hints
+	hints := p.GenerateHints(info)
+	// mutate the program according to each of the hints
+	// and triage it to check if it changes coverage
+	for _, hint := range hints {
+		newP := p.MutateWithHint(hint)
+		if newP != nil {
+			switch hint.Size {
+			case 1:
+				atomic.AddUint64(&statByteHints, 1)
+			case 2:
+				atomic.AddUint64(&statWordHints, 1)
+			case 4:
+				atomic.AddUint64(&statDWordHints, 1)
+			case 8:
+				atomic.AddUint64(&statQWordHints, 1)
+			}
+			execute(pid, env, newP, false, false, false, false, true, &statExecHints)
+		}
+	}
+}
+func execute(pid int, env *ipc.Env, p *prog.Prog, needCover, needComps, minimized, candidate, generatedByHint bool, stat *uint64) []prog.CallInfo {
 	opts := &ipc.ExecOpts{}
+	if needComps && kcovCompsEnabled {
+		// needComps has "higher priority" then needCover, so
+		opts.Flags |= ipc.FlagCollectComps
+		needCover = false
```

Comment:
Similarly, do we ever ask for needComps and needCover at the same time? If yes, then it looks like a bug, or at least needs an explanation. If not, then drop or panic.